### PR TITLE
Creating Activities Skeleton

### PR DESCRIPTION
**Pull request by brokenheart1712**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ class User < ActiveRecord::Base
            foreign_key: 'followed_id', dependent: :destroy
   has_many :following, through: :active_relationships, source: :followed
   has_many :followers, through: :passive_relationships, source: :follower
-  has_many :lessons
-  has_many :activities
+  has_many :lessons, dependent: :destroy
+  has_many :activities, dependent: :destroy
 
   validates :name, presence: true
   validates :email, confirmation: true, uniqueness: true, presence: true, format: {with: VALID_EMAIL_REGEX, message: I18n.t(:email_invalid_format)}
```
Comment (NguyenTanDuc) [inline app/models/user.rb]: remove I18n

File: app/controllers/activities_controller.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class ActivitiesController < ApplicationController
```
Comment (nguyenduyta) [inline app/controllers/activities_controller.rb]: remove this file

File: app/views/static_pages/home.html.erb [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+      </section>
+    </aside>
+    <div class="col-md-8">
+      <h3><%= t :activities %></h3>
+      <%= render 'shared/feed' %>
+    </div>
+  </div>
+<% else %>
+  <div class="center jumbotron">
+    <h1><%= I18n.t :welcome %></h1>
```
Comment (NguyenTanDuc) [inline app/views/static_pages/home.html.erb]: I18n

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -2,6 +2,6 @@ module UsersHelper
   def gravatar_for user
     gravatar_id = Digest::MD5::hexdigest user.email.downcase
     gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}"
-    image_tag gravatar_url, alt: user.name, class: "gravatar"
+    image_tag gravatar_url, alt: user.name, class: 'gravatar'
```
Comment (nguyenduyta) [inline app/helpers/users_helper.rb]: revert

File: app/models/user.rb [hunk 5]
```diff
@@ -71,6 +71,10 @@ def password_reset_expired?
     reset_sent_at < 2.hours.ago
   end
 
+  def feed
+    Activity.where 'user_id = ?', id
```
Comment (nguyenduyta) [inline app/models/user.rb]: change ' to "

File: app/controllers/activities_controller.rb (hunk 2 above)
Comment (brokenheart1712) [inline app/controllers/activities_controller.rb]: Please give me the reason?

File: app/helpers/users_helper.rb (hunk 4 above)
Comment (brokenheart1712) [inline app/helpers/users_helper.rb]: Please give me a reason?

File: app/controllers/activities_controller.rb (hunk 2 above)
Comment (nguyenduyta) [inline app/controllers/activities_controller.rb]: i don't see any action in this file.

File: app/helpers/users_helper.rb (hunk 4 above)
Comment (nguyenduyta) [inline app/helpers/users_helper.rb]: why you changed " to '. Company's convention is "

File: app/controllers/activities_controller.rb (hunk 2 above)
Comment (brokenheart1712) [inline app/controllers/activities_controller.rb]: I added action to this file :)

File: app/helpers/users_helper.rb (hunk 4 above)
Comment (brokenheart1712) [inline app/helpers/users_helper.rb]: OK. I will revert.

File: app/models/user.rb (hunk 5 above)
Comment (brokenheart1712) [inline app/models/user.rb]: OK, I will change.

File: app/controllers/activities_controller.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    @activity = current_user.activities.build(activity_params)
+    if @activity.save
+      flash[:success] = t :activity_created
+      redirect_to root_url
+    else
+      @feed_items = []
+      render 'static_pages/home'
```
Comment (nguyenduyta) [inline app/controllers/activities_controller.rb]: render "static_pages#home"

File: app/controllers/activities_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class ActivitiesController < ApplicationController
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    @activity = current_user.activities.build(activity_params)
```
Comment (nguyenduyta) [inline app/controllers/activities_controller.rb]: remove ()

File: app/controllers/activities_controller.rb (hunk 6 above)
Comment (brokenheart1712) [inline app/controllers/activities_controller.rb]: There will be error if I do that.

File: app/models/activity.rb [hunk 8]
```diff
@@ -1,3 +1,6 @@
 class Activity < ActiveRecord::Base
   belongs_to :user
+  default_scope -> {order created_at: :desc}
```
Comment (dieunb) [inline app/models/activity.rb]: slice your code

File: app/models/user.rb [hunk 9]
```diff
@@ -71,6 +71,10 @@ def password_reset_expired?
     reset_sent_at < 2.hours.ago
   end
 
+  def feed
+    Activity.where "user_id = ?", id
```
Comment (dieunb) [inline app/models/user.rb]: scope

File: app/models/user.rb (hunk 1 above)
Comment (dieunb) [inline app/models/user.rb]: slice validation

File: app/controllers/activities_controller.rb [hunk 10]
```diff
@@ -0,0 +1,19 @@
+class ActivitiesController < ApplicationController
+  before_action :logged_in_user, only: [:create]
+
+  def create
+    @activity = current_user.activities.build activity_params
+    if @activity.save
+      flash[:success] = t :activity_created
+      redirect_to root_url
+    else
+      @feed_items = []
```
Comment (dieunb) [inline app/controllers/activities_controller.rb]: ???

Comment (brokenheart1712) [inline app/controllers/activities_controller.rb]: Why do you ??? here? Please tell me the reason.

File: app/models/activity.rb [hunk 11]
```diff
@@ -1,3 +1,7 @@
 class Activity < ActiveRecord::Base
   belongs_to :user
+  validates :user_id, presence: true
+  validates :content, presence: true
+
+  default_scope -> {order created_at: :desc}
```
Comment (dieunb) [inline app/models/activity.rb]: we don't use it. Pls remove it

**Pull request closed without merging**